service/controller/chartconfig/v5/resource/chart/update.go

```diff
@@ -93,7 +93,9 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 
 	r.logger.LogCtx(ctx, "level", "debug", "message", "finding out if the chart has to be updated")
 
-	if !currentChartState.IsEmpty() && isChartModified(currentChartState, desiredChartState) {
+	isModified := !currentChartState.IsEmpty() && isChartModified(currentChartState, desiredChartState)
+	isWrongStatus := currentChartState.ReleaseStatus != desiredChartState.ReleaseStatus && !isChartInTransitionState(currentChartState)
+	if isModified || isWrongStatus {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "the chart has to be updated")
```

Comment:
Does that mean that release in state `PENDING_UPGRADE` or `DELETING` should be updated?

Reply:
Sorry it's other way around `!`.